### PR TITLE
ensures json errors are not thrown

### PR DESCRIPTION
**Pull request by bshaffer**

### CODE REVIEW

File: src/Google/Http/REST.php [hunk 1]
```diff
@@ -93,10 +94,21 @@ public static function decodeHttpResponse(
...
 
     // return raw response when "alt" is "media"
     $isJson = !($request && 'media' == $request->getQuery()->get('alt'));
-    $result = $isJson ? $response->json() : $body;
+
+    // set the result to the body if it's not set to anything else
+    if ($isJson) {
+      try {
+        $result = $response->json();
+      } catch (ParseException $e) {
```
Comment (tmatsuo) [inline src/Google/Http/REST.php]: Now the exception is just thrown away. Is it possible to log details as a warning?

Comment (bshaffer) [inline src/Google/Http/REST.php]: The assumption is the exception indicates the response *isn't* JSON, rather than indicating an actual error.

Comment (bshaffer) [inline src/Google/Http/REST.php]: Also, whatever the non-JSON is gets returned, so nothing super valuable is thrown away (IMHO), just whatever parse error was discovered by `json_decode`.

Comment (tmatsuo) [inline src/Google/Http/REST.php]: I see. Thanks.

**Comment by tmatsuo** [comment]
LGTM

**Pull request merged**